src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/Delegation/DelegatedCompletionItemResolver.cs

```diff
@@ -100,12 +100,12 @@ internal class DelegatedCompletionItemResolver : CompletionItemResolver
             return resolvedCompletionItem;
         }
 
-        if (resolvedCompletionItem.TextEdit is not null)
+        if (resolvedCompletionItem.TextEdit is { First: var textEdit })
```

Comment:
Will `First` always be non-null? I ask because elsewhere in the PR you're using `TryGetFirst`. Is that the difference between a resolved and non-resolved item?

Reply:
In other places, I grabbed your code, which is why it used `TryGetFirst(...)`. I didn't think too hard, but maybe I should. :smile:

Reply:
Oh is this based on my async completion stuff from O#?